components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -356,4 +361,18 @@ export class PrebuildManager {
         const inactiveProjectTime = 1000 * 60 * 60 * 24 * 7 * 1; // 1 week
         return now - lastUse > inactiveProjectTime;
     }
+
+    private async shouldSkipInactiveRepository(ctx: TraceContext, cloneURL: string): Promise<boolean> {
+        const span = TraceContext.startSpan("shouldSkipInactiveRepository", ctx);
+        try {
+            return (
+                (await this.workspaceDB
```

Comment:
Q: Would it make sense to add some caching here? Just in-memory, time based eviction – let's say 1d.

Thinking out loud:
* webhook events can be dense
* checking for a project seems to be a great eviction policy in general